File: frontend/src/types/football.ts
export interface FootballTeam {
  id: number;
  name: string;
  short_name: string | null;
  tla: string | null;
  crest: string | null;
}

export interface FootballCompetition {
  id: number;
  code: string;
  name: string;
  emblem: string | null;
}

export interface FootballScore {
  home: number | null;
  away: number | null;
}

export type MatchStatus =
  | "SCHEDULED"
  | "TIMED"
  | "IN_PLAY"
  | "PAUSED"
  | "FINISHED"
  | "POSTPONED"
  | "SUSPENDED"
  | "CANCELLED"
  | "UNKNOWN";

export interface FootballFixture {
  id: number;
  utc_date: string;
  status: MatchStatus;
  matchday: number | null;

  competition: FootballCompetition;
  home_team: FootballTeam;
  away_team: FootballTeam;

  full_time: FootballScore;
  half_time: FootballScore;
}

File: backend/app/models/football.py
from datetime import datetime
from enum import StrEnum

from pydantic import BaseModel


class MatchStatus(StrEnum):
    scheduled = "SCHEDULED"
    timed = "TIMED"
    in_play = "IN_PLAY"
    paused = "PAUSED"
    finished = "FINISHED"
    postponed = "POSTPONED"
    suspended = "SUSPENDED"
    cancelled = "CANCELLED"
    unknown = "UNKNOWN"


class FootballTeam(BaseModel):
    id: int
    name: str
    short_name: str | None = None
    tla: str | None = None
    crest: str | None = None


class FootballCompetition(BaseModel):
    id: int
    code: str
    name: str
    emblem: str | None = None


class FootballScore(BaseModel):
    home: int | None = None
    away: int | None = None


class FootballFixture(BaseModel):
    id: int
    utc_date: datetime
    status: MatchStatus
    matchday: int | None = None

    competition: FootballCompetition
    home_team: FootballTeam
    away_team: FootballTeam

    full_time: FootballScore
    half_time: FootballScore